cache/public/census_topic.go

```diff
@@ -0,0 +1,137 @@
+package public
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/ONSdigital/dp-frontend-homepage-controller/cache"
+	"github.com/ONSdigital/dp-topic-api/models"
+	topicCli "github.com/ONSdigital/dp-topic-api/sdk"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// UpdateCensusTopic is a function to update the census topic cache in web (public) mode.
+// This function talks to the dp-topic-api via its public endpoints to retrieve the census topic and its subtopic ids
+// The data returned by the dp-topic-api is of type *models.PublicSubtopics which is then transformed to *cache.Topic in this function for the controller
+// If an error has occurred, this is captured in log.Error and then an empty census topic is returned
+func UpdateCensusTopic(ctx context.Context, topicClient topicCli.Clienter) func() *cache.Topic {
+	return func() *cache.Topic {
+		// get root topics from dp-topic-api
+		rootTopics, err := topicClient.GetRootTopicsPublic(ctx, topicCli.Headers{})
```

Comment:
Again not sure if this call is needed, as we know what the census topic id is

Reply:
see comment above please